jmx_prometheus_javaagent/src/test/java/io/prometheus/jmx/JavaAgentIT.java

```diff
@@ -0,0 +1,103 @@
+package io.prometheus.jmx;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.Test;
+
+public class JavaAgentIT {
+    private List<URL> getClassloaderUrls() {
+        return getClassloaderUrls(getClass().getClassLoader());
+    }
+
+    private static List<URL> getClassloaderUrls(ClassLoader classLoader) {
+        if (classLoader == null) {
+            return Collections.emptyList();
+        }
+        if (!(classLoader instanceof URLClassLoader)) {
+            return getClassloaderUrls(classLoader.getParent());
+        }
+        URLClassLoader u = (URLClassLoader) classLoader;
+        List<URL> result = new ArrayList<URL>(Arrays.asList(u.getURLs()));
+        result.addAll(getClassloaderUrls(u.getParent()));
+        return result;
+    }
+
+    private String buildClasspath() {
+        StringBuilder sb = new StringBuilder();
+        for (URL url : getClassloaderUrls()) {
+            if (!url.getProtocol().equals("file")) {
+                continue;
+            }
+            if (sb.length() != 0) {
+                sb.append(':');
+            }
+            sb.append(url.getPath());
+        }
+        return sb.toString();
+    }
+
+    @Test
+    public void agentLoads() throws IOException, InterruptedException {
+        // If not starting the testcase via Maven, set the  buildDirectory and finalName system properties manually.
+        final String buildDirectory = (String) System.getProperties().get("buildDirectory");
+        final String finalName = (String) System.getProperties().get("finalName");
+        final int port = 1234;
```

Comment:
Maybe we should use http://www.mojohaus.org/build-helper-maven-plugin/reserve-network-port-mojo.html